Introduction edits (#43)
greenelab/xswap-manuscript: content/02.body.md

## Introduction

Networks contain information about relationships between entities (referred to here as "edges" between "nodes").
A node's degree is the number of edges it has in the network.

> `TODO: do we need to discuss in-degree vs out-degree and hetnet degree here?`

Reviewer: I think the discussion in the figure description should be sufficient for the introduction

Networks contain many nodes, whose degrees can be aggregated to form the network's degree distribution.

> Because different nodes can have very different degrees, real networks generally have highly variable degree distributions [@doi:10.1126/science.286.5439.509; @doi:10.1038/s41467-019-08746-5; @doi:10.1371/journal.pone.0017645].

Author: Below there is the following sentence:

```
Real networks have a variety of degree distributions (Figure {@fig:hetionet}), and they commonly exhibit degree imbalance [@doi:10.1371/journal.pone.0017645; @doi:10.1007/978-1-61779-361-5_13; @doi:10.1038/s41467-019-08746-5; @doi:10.1126/science.286.5439.509].
```

I wonder if we should introduce this concept much earlier like here?

Reviewer: I like the phrasing in this PR. Is this sentence a suggested alternative?

Author: I don;t understand this comment. The suggestion is the PR. The blockquoted sentence is from below and currently is in manuscript.

Reviewer: My bad, didn't remember that the blockquote was still included below.

> This is especially true for networks encoding biomedical knowledge or assays — where natural forces such as preferential attachment inherent to the problem domain combine with observation-based influences such as study methodology — to create non-uniform degree distributions (Figure {@fig:hetionet}).

Reviewer: Should there be spaces between the dashes? 

```suggestion
This is especially true for networks encoding biomedical knowledge or assays—where natural forces such as preferential attachment inherent to the problem domain combine with observation-based influences such as study methodology—to create non-uniform degree distributions (Figure {@fig:hetionet}).
```

Reviewer: On second thought, what do you think about the following punctuation?

```suggestion
This is especially true for networks encoding biomedical knowledge or assays, where natural forces such as preferential attachment inherent to the problem domain combine with observation-based influences such as study methodology to create non-uniform degree distributions (Figure {@fig:hetionet}).
```

Author: All suggested changes are fine are purely stylistic choices.

> ![

Author: For this figure, have we looked at including the Gene-interacts-Gene edge? Would like to show an undirected edge as a helpful reference.

Reviewer: I could include it. Compound-resembles-compound is included and is undirected

Author: My bad. Updated in https://github.com/greenelab/xswap-manuscript/pull/43/commits/b5561805ffd1edc43e16151a4306c27f6caa2024

It may be helpful to show both orientations of a bipartite edge.

Also good to use unicode separator for undirected edges. See https://git.dhimmel.com/rephetio-manuscript/#tbl:metaedges

Reviewer: I chose the metaedges to illustrate the variability of degree distributions. The final two are source/target for a single metaedge just to show that those can be different too. I'm not opposed to adding more distributions, though distributions that are similar to those already shown may not add much to the figure.

**Biomedical networks are characterized by non-uniform degree distributions.**
Seven degree distributions are plotted for six edge types Hetionet v1.0 [@doi:10.7554/eLife.26726].

> Hetionet integrates subnetworks for 24 different edge types.

Reviewer: ```suggestion
Hetionet integrates subnetworks for 24 different edge types, the degree distributions of which are analyzed separately.
```

Reviewer: I thought the added succinct-ness may be worth it here.

> To assess the degree distributions of Hetionet, each edge type is analyzed separately.

Reviewer: ```suggestion
```

Furthermore, bipartite (e.g. Anatomy→expresses→Gene) and directed (e.g. Gene→regulates→Gene) edge types have both source and target degrees that must be assessed separately.
Undirected edge types (e.g Compound–resembles–Compound) have only a single degree distribution.
Degree distributions are non-uniform and vary greatly between different networks.

> The y-axis is XX-scaled to accomodate the common occurrence where most nodes have low degree while a small portion of nodes have high degree.

Author: is this log scaled? I wonder if that is how the zero-degrees are filtered? Also is scaled rather than transformed the proper term as per the grammar of graphiques?

Reviewer: The [notebook where this figure was made](https://github.com/greenelab/xswap-analysis/blob/master/nb/3.fig1.hetionet_degree_dist/2.plot_hetionet_degree_distributions.ipynb) shows a warning message from ggplot that, "Transformation introduced infinite values in continuous y-axis." These rows (zero degree nodes) were dropped from the visualization.

Scaling is log10.

```suggestion
The vertical axis is log-10-scaled to accommodate distributions in which many nodes have low degree while a small portion have high degree.
```

## Introduction

Networks contain information about relationships between entities (referred to here as "edges" between "nodes").
A node's degree is the number of edges it has in the network.
Networks contain many nodes, whose degrees can be aggregated to form the network's degree distribution.
Because different nodes can have very different degrees, real networks generally have highly variable degree distributions [@doi:10.1126/science.286.5439.509; @doi:10.1038/s41467-019-08746-5; @doi:10.1371/journal.pone.0017645].
This is especially true for networks encoding biomedical knowledge or assays, where natural forces such as preferential attachment inherent to the problem domain combine with observation-based influences such as study methodology to create non-uniform degree distributions (Figure {@fig:hetionet}).

![
**Biomedical networks are characterized by non-uniform degree distributions.**

> Seven degree distributions are plotted for six edge types Hetionet v1.0 [@doi:10.7554/eLife.26726].

Reviewer: Figure now has more distributions
```suggestion
Eight degree distributions are plotted for six edge types Hetionet v1.0 [@doi:10.7554/eLife.26726].
```

Hetionet integrates subnetworks for 24 different edge types, the degree distributions of which are analyzed separately.
Furthermore, bipartite (e.g. Anatomy→expresses→Gene) and directed (e.g. Gene→regulates→Gene) edge types have both source and target degrees that must be assessed separately.
Undirected edge types (e.g Compound–resembles–Compound) have only a single degree distribution.
Degree distributions are non-uniform and vary greatly between different networks.

> The y-axis is log~10~-scaled to accomodate the common occurrence where most nodes have low degree while a small portion of nodes have high degree.

Reviewer: What does the `~` do?

Author: subscript

> Several distributions have node degree ceilings corresponding to a nodes being connected to all other possible nodes.

Reviewer: ```suggestion
Several distributions have node degree ceilings corresponding to a node being connected to all other possible nodes.
```
